### DISCUSSION TITLE
Victron 150/60 VE.direct probleme

### DISCUSSION THREAD

**Discussion by bikerbiker12** [hoylabs/OpenDTU-OnBattery#551] [Q&A]
Hallo,
Habe heute meinen Laderegler 100/30 gegen 150/60 getauscht! Nun habe ich in der LIvE Ansicht ein rot/blaues blinken beim MPPT Regler! Habe nur das VE.direct Kabel umgesteckt und sonst nichts geändert! Hat jemand eine Idee und kann mir helfen!? 
Grüße Marco 
![IMG_7330](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/126656972/2982dcfc-9539-4b60-a047-e3e7e3e1983b)
![IMG_7331](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/126656972/5b1575f3-306b-4546-914c-707ba76753a7)

**Comment by schlimmchen** [comment]
Ist der neue Regler vielleicht gebraucht und hat eine alte Firmware? Oder ist der VE.Direct Anschluss falsch eingestellt? Was steht denn im Log wenn du Verbose Logging einschaltest?

**Reply by bikerbiker12** [reply]
Der Anschluss steht auf TX normale Kommunikation! 
mit dem Logging musst du mir mal helfen?! 🙃

**Reply by bikerbiker12** [reply]
Achja, der Regler ist neu und hat alle Updates!

**Comment by Manos1966** [comment]
Das letzte Bild zeigt SmartSolar MPPT 150|60 
Das bedeutet, die OpenDTU hat die Verbindung gemacht.

Vielleicht das Problem liegt beim Satz: 
Grund fuer das Ausschalten: No Input Power (Kein Power im Eingang )

**Reply by bikerbiker12** [reply]
Das steht da sobald keine Sonnenergie mehr anliegt! 
Es kommt Daten an, im gleichen Abstand wie das blinken ist, dann wieder null!

**Comment by madmartin** [comment]
Die Victron Lader senden über das VE.direct interface alle 1-2 Sekunden einen Datensatz. Wann die OpenDTU das empfangen hat steht bei "letzte Aktualisierung vor X sekunden". Wenn das länger als 8? Sekunden her ist wird die Farbe des Balkens von blau nach rot geändert.
Dein Problem ist also daß die OpenDTU nicht regelmässig aktuelle Daten über die VEdirect Schnittstelle reinbekommt.
Woran das liegt? 
Das kann alles mögliche sein.
Fragen: Wie sieht die Verbindung zum Lader genau aus? Durchgehendes Kabel? Hast Du einen Pegelwandler dazwischen, oder einen ADUM1201?
Hast Du die Hinweise im [Wiki](https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/VE.Direct-Interface)  gelesen und befolgt?

**Reply by bikerbiker12** [reply]
Die Anzeige wechselt im Sekundentakt! Am 100/30 hat es funktioniert daher habe ich es erst mal so belassen! Das Kabel ist mit GND und TX direkt ohne Levelshifter oder ADUM1201 angeschlossen, die beiden anderen sind nicht angeschlossen!

**Comment by Manos1966** [comment]
Wenn der Victron MPPT kein Strom hat, ist er nicht aus? = sendet keine Daten?

"Made by design" 

Ich wurde behaupten, deine OpenDTU sowie dein Kabel funktionieren einwandfrei.

Aber aus irgendeinem Grund, No Input Power (Kein Power im Eingang ), etwas stimmt nicht mit deinen PV Paneele.

**Reply by bikerbiker12** [reply]
Zu dem Zeitpunkt wurde über den Akku eingespeist und der Victron automatisch mitversorgt! Null Power betitelt den PV Eingang, da kam natürlich nichts rein abends!

**Reply by madmartin** [reply]
So sehe ich das auch. @Manos1966 hat übersehen daß das "Gehirn" vom Victron Lader über die Batterie versorgt wird.

Also ich vermute mal daß die Pegel bei dem neuen Lader etwas anders sind als beim alten und daher sich nun die ungleichen Spannungen an der Schnittstelle bemerkbar machen. Es ist ohnehin leichtsinnig das ohne Pegelwandler  auf Dauer zu betreiben. Ist eh ein Wunder daß der ESP32 das aushält.  Bei einem RasPi würdest Du auf diese Weise (5V input an GPIO) sofort die ganze GPIO-Bank zerstören.

**Comment by Manos1966** [comment]
Also, nur zur Info, 
ich habe eine Test-Installation die seit May 2023 läuft mit einem Victron 150/35 und ohne ADUM1201/Levelshifter.
Ich wollte wissen, wie Lange der "Direkte Anschluss" halten wird. Keine Probleme bis jetzt (bin ich selber überrascht).


Jetzt zum Fall hier. Ich bin tatsaechlich davon ausgegangen, dass der Victron nicht genug Power weder von den Paneelen, noch von der Batterie bekommt.

Anbei ein Bild meines 150/35 von heute Nacht. Er hat Firmware 161
Vergleicht die Infos von bikerbiker12s Bilder:
Status Ladeausgang ist bei mir ON (bei ihm OFF)
Grund fuer das Ausschalten ist bei mir No input Power (bei ihm ein Mal "No input Power", ein Mal "Not Off")

![Untitled](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/06c04e26-2791-4406-84d3-cdfdead5a104)

@bikerbiker12 Was zeigt dein Fenster Ausgang fuer Spannung (meiner im Bild 49,14V)

Jemand hatte das gleiche Problem, die Loesung war unerwartet einfach (Kabel):
VE.Direct: irregular and long periods without update #411

**Comment by bikerbiker12** [comment]
Status Ausgang wird bei mir im Wechsel die Spannung der Batterie angezeigt, oder eben nichts! Ich prüfe heute noch mal die Kabelverbindungen! 
Per MQQT kommen im gleichen Rhythmus Daten an oder nicht! 
Der PV Gesamtertrag kommt nur manchmal sporadisch

**Reply by Manos1966** [reply]
OK, dann die Parametern einzeln ausschliessen:
Entweder hat dein Victron kein Strom (PV/Batterie), 
ODER bekommt er den Befehl zum NEUSTART ständig vom Kabel. Da du erzählst hast, du hast nur TX & GND von VICTRON-Seite (stimmt das? Bitte prüfe, dass du wirklich TX & GND angeschlossen hast, manche Victrons haben den Stecker umgekehrt) sollte es gar nicht der Fall sein. 

-PV-Seite trennen (ich hoffe du hast Leistungstrennschalter)
-VE.Direkt Kabel Entfernen 
-Nur Batterie Angeschlossen lassen

- Bleibt die Verbindung stabil und die Spannung der Batterie angezeigt?
- Falls ja, die PV Seite auch anschließen
- Bleibt die Verbindung stabil und beide Seiten (PV-Eingang/Batterie-Ausgang) funktionieren?
- Falls ja, dann bleibt dein VE.Direkt Kabel ;-)

**Comment by bikerbiker12** [comment]
Hallo,
also der Victron wird mit Strom versorgt und ist auch per Bluetooth erreichbar, die Status-LED blinkt regelmäßig wie sie es soll,
Die Verbindung habe ich noch einmal überprüft (am Shunt und am vorhergehenden 100/30) ist die Verbindung da. 
Das VE.Direkt Kabel ist ok, habe ich nocheinmal durchgemessen und es kommen an der Seite der DTU auch ca. 3,15-3,3V an. 
Leider habe ich den 100/30 nicht mehr, sonst hätte ich diesen noch einmal angeschlossen.

**Reply by Manos1966** [reply]
> Das VE.Direkt Kabel ist ok, habe ich nocheinmal durchgemessen

OK, dann: OHNE VE.Direkt KAbel angeschlossen, funktioniert alles korrekt ueber laengere Zeit?

**Reply by bikerbiker12** [reply]
Ja, alles bestens ohne Kabel. Ich habe sogar noch einen SmartShunt über VE angeschlossen. Dieser läuft ohne Probleme das VE Kabel vom Regler hatte ich ebenfalls getauscht. Ich vermute, dass der Lade Regler die Daten pusht und dazwischen die Stromversorgung unterbricht

**Reply by Manos1966** [reply]
OK, das ist auch eine gute Nachricht.

Das bedeutet, der Victron funktioniert einwandfrei. 
Es haengt nur an den VE.Direkt Einstellungen 

Da du dich gut mit dem 100/30 auskennst, hast du auch das Handbuch des 150/35 gelesen? 😈 
Und ist deine TX Einstellung auf "Normal"
![Capture](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/99e40813-716a-4167-b12d-3b697c9a9c73)

**Reply by bikerbiker12** [reply]
Ja, aber selbstverständlich hab ich das gelesen. Ja die Einstellungen habe ich bereits kontrolliert steht auf normale Kommunikation.

**Reply by Manos1966** [reply]
> Ja, aber selbstverständlich hab ich das gelesen. Ja die Einstellungen habe ich bereits kontrolliert steht auf normale Kommunikation.

👍 
Du willst es uns nicht einfach machen.... 🤣 

Und du hast NUR die zwei Kabel (GND, TX) angeschlossen, die restlichen (RX,5V) sind nicht angeschlossen?
![Capture](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/d39d8ff8-a8bd-4425-b195-1313ecb27956)

**Reply by bikerbiker12** [reply]
Ja nur die beiden (GND + TX). Ist mal eine Denksportaufgabe vor Weihnachten 😂

**Reply by Manos1966** [reply]
Verdammt! 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️

**Comment by bikerbiker12** [comment]
Ich hab nun noch den SmartShunt mit dem MPPT Anschluss getauscht und das PinMapping ausgetauscht, das gleiche Spiel! Ich kann somit einen Defekt am GPIO der DTU ausschließen, Kabel etc ebenso. 
Habe auf dem MPPT ältere Firmware Versionen aufgespielt, alles ohne Erfolg. Es kann also nur direkt am MPPT liegen oder an der Programmierung der DTU.

**Comment by Manos1966** [comment]
Hmmm... Interessant, 

es ist bei mir auch passiert: 17:45
![Victron 150-30 Problem](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/ed9054c3-6063-4b9c-a3f0-a57a56e0e47a)

**Comment by bikerbiker12** [comment]
ich wars nicht.....:-)

**Comment by Manos1966** [comment]
Hallo @bikerbiker12  wirft mal ein Blick hier
[heute funktioniert der VICTRON perfekt und die Updates kommen im 1-2 Sekunden-Takt](https://github.com/helgeerbe/OpenDTU-OnBattery/discussions/557#discussioncomment-7919817)

Wieviel Volt zeigt dein Eingang? (da wo ich 57,73V habe)

**Reply by bikerbiker12** [reply]
Servus,
Bei mir steht bei PV Eingang ca 60V. Spielt aber bei mir keine Rolle, egal wieviel Spannung anliegt habe ich das On/Off Bild!

**Comment by bikerbiker12** [comment]
Ich habe heute mal den Laderegler gegen einen neuen (gleicher Typ) getauscht! Leider ohne Erfolg! 
das aktuelle Release kann ich nicht hochladen (Bad request)

**Reply by Manos1966** [reply]
Hallo @bikerbiker12 
wirf ein Auge hier:
[Kommentare](https://github.com/helgeerbe/OpenDTU-OnBattery/discussions/557#discussioncomment-7948490)

Hast du selber deine OpenDTU gebaut, oder gekauft?

**Reply by bikerbiker12** [reply]
Ich hab meine DTU von Nordskills fertig gekauft! Ich verfolge deinen Thread mit, ich habe allerdings etwas andere Symptome! Bei geht das rund um die Uhr so, solange der Victron irgendwie mit Strom versorgt wird! 

https://github.com/helgeerbe/OpenDTU-OnBattery/assets/126656972/6e5d6ac3-5d23-4e7d-88bd-26ff7c4308e0

**Comment by Manos1966** [comment]
bist du in der naehe von Bonn?

Vielleicht solltest du eine andere OpenDTU probieren.

**Reply by bikerbiker12** [reply]
Nein leider nicht! 
meinst du das es an de DTU liegen kann? Aberauf Verdacht Geld investieren…?!

**Reply by Manos1966** [reply]
Na ja, ich versuche die logische Schritte zu finden....

1. Hast du ein Raspberry? Dann nutze die Victron Simulation um deine OpenDTU zu testen. Du brauchts nichts mehr als ein Kabel.
[victron-vedirect-emulator-for-linux](https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Victron-MPPT-Tipps#victron-vedirect-emulator-for-linux)

Du brauchts nichts mehr als ein Kabel auf die zwei Raspi PINs
https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/00c503ce-90da-44d1-8a82-fa59686ea357

**Comment by ymca72** [comment]
Hallo,

ich hänge mich mal da dran, weil ich habe das gleiche problem habe wie bikerbiker12.

Ich bin gerade dran meine PV-Anlage Stück für Stück in Betrieb zu nehmen. Wollte heute an sich die Panels aufs Dach bringen aber der Regen hat mir ein Strich durch die Rechnung gemacht. 

Also dann statt dessen heute meine OPEN-DTU aufgebaut mit ADUM1201 an einem 150/85. Alles auf den neusten Firmwarestand. Batterie zeigt über die App 52,73V an (50%). Von oben kann noch nix kommen, weil nix da ist. Verbunden habe ich VE-direkt TX an die DTU über GPIO3 als RX. 5V und GND kommen auf der einen Seite an den ADUM (VE-Seite) und 3,3V und GND vom ESP. TX vom VE auf RX GPIO3. Daten kommen vom Laderegler aber auch bei mir schaltet die im Sekundentakt von Blau auf Rot. Ich sehe die Daten kurz und dann sie wieder weg.

Hat jemand schon ne Lösung für das "pumpen" ?

Achja, ein HM600 liegt hier auch schon rum ist aber noch nicht angeschlossen, weil ich erstmal die Batterie aufladen will bevor der WR dran kommt. Wie gesagt, ich nehme gerade scheibenweise die Sachen in Betrieb und das nervt gerade etwas.

Danke und Gruß

**Reply by ymca72** [reply]
Hallo,
ich habe mal weiter rumexperimentiert und auch weiter geschaut, ob jemand ne Lösung dafür hat. Mir ist sponatn eingefallen, das der NRF24 ja auch 3,3V liegt und habe heuteauf den 3,3V mal 100µF zum stabilisieren drangehängt, hat aber nix gebraucht. Ich habe auch in einem anderen Thread gelesen, das die Leitungslänge wohl eine Rolle spielt und auch das habe ich gestestet aber auch hier, nachdem ich die Kabel so kurz wie möglich gehalten habe, schaltet es immer noch zwischen on und off her. 
In der Konsole scheint mir, das die Datenpakete immer wieder abgehackt werden, als würde der ESp32 nicht mit der Arbeit zurecht kommen. Ich abb e auch mal den NRF24 mal weggenommen aber auch hier wurde die stabilität nicht besser.

Hat noch jemand ne Idee?

Danke und Gruß

**Reply by Manos1966** [reply]
Hallo @ymca72 
welche ESP32 Platine hast du, welchen USB Anschluss und welches WiFi Modul hat sie drauf?

**Reply by ymca72** [reply]
Hallo,

einen ESP-Wroom-32, selbst aus China direkt importiert. Hatte auch deswegen mit den Modulen nie Probleme. VE.direkt ist an GPIO3 gemappt

![Screenshot_20240218_212421_Opera 1](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/79200928/85a253ef-b8c3-49b8-9bec-e6c7234158cc)
![20240218_212913 1](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/79200928/1fe087c4-0436-46ad-934d-3daaa7d7a0ff)

**Comment by Manos1966** [comment]
Micro-USB Anschluss, oder USB-C ?
WiFi Modul CP2102 oder CH340?

**Reply by ymca72** [reply]
Ich bevorzuge die Module mit 2102 Chipsatz und Micro-USB wegen besseren Verfügbarkeit. Hatte anfangs etwas schlechte Erfahrung mit 340 gehabt und deswegen bevorzugt 2102. Aber was hat der Stecker und der USB-Chipsatz Relevanz mit dem Betrieb des ESP32? Weil der Stecker wird nur noch zur Stromversorgung benutzt und selbst das könnte umgangen werden wenn man wollte.

**Reply by Manos1966** [reply]
OK, also es ist CP2102 und Micro-USB 
Dann, habe ich keine Erklaerung warum du so viele Probleme hast.

**Comment by poolcat4711** [comment]
Hallo zusammen,

das Problem mit dem "pumpen" hat mein 250/85 seit der ersten Stunde und es hat mich eigentlich nie interessiert, da die wichtigen Werte per MQTT übermittelt wurden. Allerdings eben nicht alle - die H Werte fehlen. Ich habe mir das ganze eben mal kurz angeschaut. Der VE.Direct übermittelt folgende Sachen:

```
22:02:39.323 > [VE.Direct] serial input (161 Bytes):
22:02:39.386 > [VE.Direct] 3a 41 32 30 30 31 30 30 33 43 46 36 30 44 30 30
22:02:39.447 > [VE.Direct] 45 42 0a 3a 41 31 32 45 45 30 30 33 45 37 30 39
22:02:39.500 > [VE.Direct] 44 0a 3a 41 44 42 45 44 30 30 38 42 30 35 46 33
22:02:39.554 > [VE.Direct] 0a 3a 41 42 42 45 44 30 30 38 32 30 31 32 30 0a
22:02:39.635 > [VE.Direct] 0d 0a 45 52 52 09 30 0d 0a 4c 4f 41 44 09 4f 4e
22:02:39.726 > [VE.Direct] 0d 0a 52 65 6c 61 79 09 4f 46 46 0d 0a 48 31 39
22:02:39.819 > [VE.Direct] 09 31 31 35 38 31 31 0d 0a 48 32 30 09 31 32 34
22:02:39.875 > [VE.Direct] 0d 0a 48 32 31 09 39 32 32 0d 0a 48 32 32 09 32
22:02:40.016 > [VE.Direct] 32 34 0d 0a 48 32 33 09 35 39 36 0d 0a 48 53 44
22:02:40.066 > [VE.Direct] 53 09 32 35 36 0d 0a 43 68 65 63 6b 73 75 6d 09
22:02:40.158 > [VE.Direct] 0f
22:02:40.222 > [Victron MPPT] Text Event PID: Value: 0XA116
22:02:40.277 > [Victron MPPT] Text Event FWE: Value: 313FF
22:02:40.338 > [Victron MPPT] Text Event SER: Value: HQ2316VWCFY
22:02:40.407 > [Victron MPPT] Text Event V: Value: 49250
22:02:40.544 > [Victron MPPT] Text Event I: Value: 0
22:02:40.653 > [Victron MPPT] Text Event VPV: Value: 3730
22:02:40.755 > [Victron MPPT] Text Event PPV: Value: 0
22:02:40.850 > [Victron MPPT] Text Event MPPT: Value: 0
22:02:40.955 > [Victron MPPT] Text Event CS: Value: 0
22:02:41.041 > [Victron MPPT] Text Event OR: Value: 0X00000001
22:02:41.158 > [VE.Direct] serial input (177 Bytes):
22:02:41.218 > [VE.Direct] 3a 41 32 30 30 31 30 30 33 44 46 36 30 44 30 30
22:02:41.275 > [VE.Direct] 45 41 0a 3a 41 31 32 45 45 30 30 33 41 37 30 41
22:02:41.348 > [VE.Direct] 31 0a 3a 41 44 42 45 44 30 30 38 37 30 35 46 37
22:02:41.402 > [VE.Direct] 0a 3a 41 42 42 45 44 30 30 37 34 30 31 32 45 0a
22:02:41.459 > [VE.Direct] 0d 0a 50 49 44 09 30 78 41 31 31 36 0d 0a 46 57
22:02:41.556 > [VE.Direct] 45 09 33 31 33 46 46 0d 0a 53 45 52 23 09 48 51
22:02:41.655 > [VE.Direct] 32 33 31 36 56 57 43 46 59 0d 0a 56 09 34 39 32
22:02:41.710 > [VE.Direct] 35 30 0d 0a 49 09 30 0d 0a 56 50 56 09 33 37 33
22:02:41.880 > [VE.Direct] 30 0d 0a 50 50 56 09 30 0d 0a 4d 50 50 54 09 30
22:02:41.964 > [VE.Direct] 0d 0a 43 53 09 30 0d 0a 4f 52 09 30 78 30 30 30
22:02:42.022 > [VE.Direct] 30 30 30 30 31 0d 0a 43 68 65 63 6b 73 75 6d 09
22:02:42.106 > [VE.Direct] 46
22:02:42.288 > [Victron MPPT] Text Event ERR: Value: 0
22:02:42.386 > [Victron MPPT] Text Event LOAD: Value: ON
22:02:42.473 > [Victron MPPT] Text Event RELAY: Value: OFF
22:02:42.578 > [Victron MPPT] Text Event H19: Value: 115811
22:02:42.694 > [Victron MPPT] Text Event H20: Value: 124
22:02:42.804 > [Victron MPPT] Text Event H21: Value: 922
22:02:42.904 > [Victron MPPT] Text Event H22: Value: 224
22:02:43.107 > [Victron MPPT] Text Event H23: Value: 596
22:02:43.208 > [Victron MPPT] Text Event HSDS: Value: 256
```

Also ein Paket mit den interessanten Werten, ein weiteres für die anderen. Im VeDirectFrameHandler steht, das ein gültiger Frame eine korrekte Checksumme an der richtigen Stelle haben muss. (Die Funktion die auch das vorhandensein einer Seriennummer im Frame prüft, wird nicht benutzt. Wär hier aber auch falsch (--> isDataValid)). Beide Pakete sind also gültig und werden weiterverarbeitet.

Jetzt kommt der Clou. Ein Frame wird nach korrektem Empfang an openDTU übergeben und der Empfangspuffer wieder zurückgesetzt (initialisiert). Dadurch werden alle Werte standardmäßig wieder auf 0 gesetzt. Sachen, die also nicht im Paket drinstehen (im 2ten bspw. Ladeausgang, Spannung, Zustand usw) sind in dem ebenfalls korrektem zweiten Paket - 0. Sprich beim ersten Paket ist der Statusbalken blau, beim zweiten Rot. Beim dritten wieder blau und so weiter.

Nimmt man aus dem Quelltext die Initialisierung des Zwischenspeichers raus - dass sich die Werte also immer nur fortlaufend aktualisieren, bleibt alles stabil blau und meine H19-H23 Werte sind auch im MQTT.

Die entsprechende Zeile (wiegesagt, quick und dirty) auskommentieren und alles bleibt blau: 

lib\VeDirectFrameHandler\VeDirectMpptController.cpp:84
	//_tmpFrame = {};

@schlimmchen - ich bin leider ein absoluter git noob und kenne auch die weiteren Implikationen der Änderung nicht. Vielleicht schaut sich das mal einer von euch an. Bei mir geht aber alles wie es soll damit =)

Edit: Was mir gerade noch einfällt - man müsste natürlich beim Start ein paar Pakete abwarten - da ja sonst bspw. der Gesamtertrag (nach dem Start, nach dem ersten Paket dass den Wert ja nicht enthält) mit 0 gemeldet werden würde.

**Reply by schlimmchen** [reply]
Guter Fund! Weiter in #689.
